app/assets/javascripts/discourse/app/lib/uppy-checksum-plugin.js

```diff
@@ -0,0 +1,93 @@
+import { Plugin } from "@uppy/core";
+import { Promise } from "rsvp";
+
+export default class UppyChecksum extends Plugin {
+  constructor(uppy, opts) {
+    super(uppy, opts);
+    opts = opts || {};
+    this.id = opts.id || "uppy-checksum";
+    this.capabilities = opts.capabilities;
+
+    // this is meaningless, can be anything
+    this.type = "preprocessor";
+  }
+
+  canUseSubtleCrypto() {
+    if (!window.isSecureContext) {
+      this.warn(
+        "Cannot generate cryptographic digests in an insecure context (not HTTPS)."
+      );
+      return false;
+    }
+    if (this.capabilities.isIE11) {
```

Comment:
Is this only IE11 or should it also include IE versions before it?

Reply:
I think it's fine, discourse-encrypt does the same thing and we check for the other capabilities in the next line. This whole function is about checking for old browsers.

https://github.com/discourse/discourse/blob/7f2a571817062f5bd9814d8f43b1ab903f9fd470/app/assets/javascripts/discourse/app/lib/uppy-checksum-plugin.js#L26-L33